l10n_it_delivery_note_inter_warehouse/wizards/delivery_note_select.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!--
+  Copyright 2023 Ooops404
+  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+-->
+<odoo>
+    <record id="stock_delivery_note_select_form_view" model="ir.ui.view">
+        <field name="name">stock.delivery.note.select.form</field>
+        <field name="model">stock.delivery.note.select.wizard</field>
+        <field
+            name="inherit_id"
+            ref="l10n_it_delivery_note.stock_delivery_note_select_form_view"
+        />
+        <field name="arch" type="xml">
+            <field name="selected_picking_ids" position="after">
+                <field name="is_inter_warehouse" invisible="1" />
+                <p
+                    class="text-center alert alert-warning"
+                    role="alert"
+                    attrs="{'invisible': ['|', ('delivery_note_id', '=', False), ('is_inter_warehouse', '=', False)]}"
+                >
+                    <span class="fa fa-warning" />
+                    Warning: Selected Delivery Note's data will be overwritten based on the current picking.
+                </p>
+            </field>
+        </field>
+    </record>
+</odoo>
```

Comment:
**issue**: you intend to add this warning here as well as in https://github.com/OCA/l10n-italy/pull/3770 which would mean it'd show up twice for people who have both modules installed.

You may want to add it in the base `l10n_it_delivery_note` module instead.

Also, I'd only show the warning if there is more than one picking. And in the message, instead of `the current picking` I would specificy `the last picking`

Reply:
I don't totally agree with that, the domains to make it invisible are different based on the feature itself, so only one warning message will appear at the time based on the conditions of the pickings. 

Haven't tested it yet but, isn't it better to keep it separated since, as similar as they look, they serve different purpose?

(Changed the message as you suggested tho, thanks!)